### PR TITLE
rules yaml files updated as per yaml linting conditions

### PR DESCRIPTION
**Pull request by h4l0gen**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind feature

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

/kind failing-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area rules

> /area registry

> /area build

> /area documentation

**Proposed rule [maturity level](https://github.com/falcosecurity/rules/blob/main/CONTRIBUTING.md#maturity-levels)**

> Uncomment one (or more) `/area <>` lines (only for PRs that add or modify rules):

> /area maturity-stable

> /area maturity-incubating

> /area maturity-sandbox

> /area maturity-deprecated

**What this PR does / why we need it**:
to pass yaml lint workflow test.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by h4l0gen** [comment]
Yaml_lint may fail on some lines. Am looking into them, and fix it ASAP!

**Comment by h4l0gen** [comment]
Hey @FedeDP @Andreagit97 @incertum @leogr , finally i 🟢 YAML-Lint check for all rules yaml files🙃. But some other checks still ❌ and am not getting idea to solve it, please give me idea 🥲.

**Comment by h4l0gen** [comment]
@FedeDP @leogr as per this [discussion ](https://github.com/falcosecurity/rules/pull/234#issuecomment-2015241114) I updated all rules files. Now our CI is completely happy 💚

File: rules/falco-deprecated_rules.yaml [hunk 1]
```diff
@@ -110,68 +134,93 @@
...
+    Disallowed outbound connection destination
+    (connection=%fd.name lport=%fd.lport rport=%fd.rport fd_type=%fd.type
+    fd_proto=fd.l4proto evt_type=%evt.type user=%user.name user_uid=%user.uid
+    user_loginuid=%user.loginuid process=%proc.name proc_exepath=%proc.exepath
+    parent=%proc.pname command=%proc.cmdline terminal=%proc.tty
+    %container.info)
+
   priority: NOTICE
-  tags: [maturity_deprecated, host, container, network, mitre_command_and_control, TA0011]
+  tags: [maturity_deprecated, host, container, network,
```
Comment (incertum) [inline rules/falco-deprecated_rules.yaml:159]: Tags now seem formatted inconsistently, for example few lines above we have:

```
tags: [
    maturity_deprecated, host, container, network,
    mitre_latera_movement, T1021.004
  ]
```

File: rules/falco-deprecated_rules.yaml [hunk 2]
```diff
@@ -110,68 +134,93 @@
...
-    outbound 
+    outbound
     and not ((fd.sip in (allowed_outbound_destination_ipaddrs)) or
-             (fd.snet in (allowed_outbound_destination_networks)) or
-             (fd.sip.name in (allowed_outbound_destination_domains)))
+            (fd.snet in (allowed_outbound_destination_networks)) or
+            (fd.sip.name in (allowed_outbound_destination_domains)))
   enabled: false
-  output: Disallowed outbound connection destination (connection=%fd.name lport=%fd.lport rport=%fd.rport fd_type=%fd.type fd_proto=fd.l4proto evt_type=%evt.type user=%user.name user_uid=%user.uid user_loginuid=%user.loginuid process=%proc.name proc_exepath=%proc.exepath parent=%proc.pname command=%proc.cmdline terminal=%proc.tty %container.info)
+  output: >
```
Comment (incertum) [inline rules/falco-deprecated_rules.yaml:150]: @leogr you specifically didn't or don't like the `>` in general. What is is going to be going forward?

File: rules/falco-deprecated_rules.yaml [hunk 3]
```diff
@@ -15,23 +15,35 @@
...
+# Proposal: Rules Adoption Management Maturity Framework
+# URL: https://github.com/falcosecurity/rules/blob/main/
+# proposals/20230605-rules-adoption-management-maturity-framework.md
+# `tags` fields also include information about the type of
+# workload inspection (host and/or container), and Mitre Attack killchain
+# phases and Mitre TTP code(s)
 # Mitre Attack References:
 # [1] https://attack.mitre.org/tactics/enterprise/
-# [2] https://raw.githubusercontent.com/mitre/cti/master/enterprise-attack/enterprise-attack.json
+# [2] https://raw.githubusercontent.com/mitre/cti/master/
```
Comment (incertum) [inline rules/falco-deprecated_rules.yaml:30]: Is it necessary to split up links to different lines?

File: rules/falco-incubating_rules.yaml [hunk 4]
```diff
@@ -16,40 +16,61 @@
...
 - macro: open_write
-  condition: (evt.type in (open,openat,openat2) and evt.is_open_write=true and fd.typechar='f' and fd.num>=0)
+  condition: (evt.type in (open,openat,openat2) and evt.is_open_write=true and
+     fd.typechar='f' and fd.num>=0)
 
 - macro: open_read
-  condition: (evt.type in (open,openat,openat2) and evt.is_open_read=true and fd.typechar='f' and fd.num>=0)
+  condition: >
+    (evt.type in
+      (
```
Comment (incertum) [inline rules/falco-incubating_rules.yaml:46]: Personally not a fan of this formatting here and it's inconsistent with above's `open_write`.

File: rules/falco-incubating_rules.yaml [hunk 5]
```diff
@@ -102,38 +123,52 @@
...
 - list: shadowutils_binaries
   items: [
     chage, gpasswd, lastlog, newgrp, sg, adduser, deluser, chpasswd,
-    groupadd, groupdel, addgroup, delgroup, groupmems, groupmod, grpck, grpconv, grpunconv,
-    newusers, pwck, pwconv, pwunconv, useradd, userdel, usermod, vigr, vipw, unix_chkpwd
-    ]
+    groupadd, groupdel, addgroup, delgroup, groupmems, groupmod, grpck,
+    grpconv, grpunconv, newusers, pwck, pwconv, pwunconv, useradd,
+    userdel, usermod, vigr, vipw, unix_chkpwd
+  ]
```
Comment (incertum) [inline rules/falco-incubating_rules.yaml:164]: Same feedback -- I think we should pick one style for lists.

File: rules/falco-deprecated_rules.yaml (hunk 3 above)
Comment (h4l0gen) [reply rules/falco-deprecated_rules.yaml:30]: @incertum we have constraint for line-length <= 80 chars. so i have to split them

File: rules/falco-deprecated_rules.yaml (hunk 1 above)
Comment (h4l0gen) [reply rules/falco-deprecated_rules.yaml]: Hey @incertum tags format is updated as per your suggestion.

File: rules/falco-incubating_rules.yaml (hunk 4 above)
Comment (h4l0gen) [reply rules/falco-incubating_rules.yaml]: @incertum i changed this formatting.

File: rules/falco-incubating_rules.yaml (hunk 5 above)
Comment (h4l0gen) [reply rules/falco-incubating_rules.yaml:161]: yes @incertum i changed this to format to same as tags and is consistent in all rule files

**Review by leogr — COMMENTED** [review]
/hold

~~I've just realized that with this change, we will put the `output:` on multiple lines, and I'm not sure that is what we want.~~

I was wrong, sorry

**Comment by h4l0gen** [comment]
Hi @leogr, what are your thoughts on this PR? Have we made any decisions yet? Feel free to assign me any related tasks; I would love to work on them.

**Comment by leogr** [comment]
> Hi @leogr, what are your thoughts on this PR? Have we made any decisions yet? Feel free to assign me any related tasks; I would love to work on them.

I have only two points:

1. Line-width: 80 characters are too few, IMO. I'd propose 120 or more (@falcosecurity/rules-maintainers we need to reach a consensus on this -  cc @falcosecurity/core-maintainers your opinion is welcome).
2. Using `>` in yaml should be avoided since it appends an unwanted new line at the end (see example below)


```yaml
- example1: 
    bla bla bla
  example2: >
    bla bla bla
```
:point_down: 
```json
[
    {
        "example1": "bla bla bla",
        "example2": "bla bla bla\n"
    }
]
```

@h4l0gen, you can proceed to update this PR if you want to (it may be useful as a preview of the end result), but we need to reach a consensus on the style before merging it.

**Comment by leogr** [comment]
> > Hi @leogr, what are your thoughts on this PR? Have we made any decisions yet? Feel free to assign me any related tasks; I would love to work on them.
> 
> I have only two points:
> 
> 1. Line-width: 80 characters are too few, IMO. I'd propose 120 or more (@falcosecurity/rules-maintainers we need to reach a consensus on this -  cc @falcosecurity/core-maintainers your opinion is welcome).
> 2. Using `>` in yaml should be avoided since it appends an unwanted new line at the end (see example below)
> 
> ```yaml
> - example1: 
>     bla bla bla
>   example2: >
>     bla bla bla
> ```
> 
> 👇
> 
> ```json
> [
>     {
>         "example1": "bla bla bla",
>         "example2": "bla bla bla\n"
>     }
> ]
> ```
> 
> @h4l0gen, you can proceed to update this PR if you want to (it may be useful as a preview of the end result), but we need to reach a consensus on the style before merging it.

@falcosecurity/rules-maintainers @darryk10 @loresuso any thoughts on my above proposal?

**Comment by darryk10** [comment]
> > > Hi @leogr, what are your thoughts on this PR? Have we made any decisions yet? Feel free to assign me any related tasks; I would love to work on them.
> > 
> > 
> > I have only two points:
> > 
> > 1. Line-width: 80 characters are too few, IMO. I'd propose 120 or more (@falcosecurity/rules-maintainers we need to reach a consensus on this -  cc @falcosecurity/core-maintainers your opinion is welcome).
> > 2. Using `>` in yaml should be avoided since it appends an unwanted new line at the end (see example below)
> > 
> > ```yaml
> > - example1: 
> >     bla bla bla
> >   example2: >
> >     bla bla bla
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 👇
> > ```json
> > [
> >     {
> >         "example1": "bla bla bla",
> >         "example2": "bla bla bla\n"
> >     }
> > ]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @h4l0gen, you can proceed to update this PR if you want to (it may be useful as a preview of the end result), but we need to reach a consensus on the style before merging it.
> 
> @falcosecurity/rules-maintainers @darryk10 @loresuso any thoughts on my above proposal?

I agree with both points. 
If we can remove the `< `, it would simplify the overall syntax, and it would be easier to maintain.

**Comment by h4l0gen** [comment]
@leogr If we increase line length value, we will solve many issues of formatting, as I've accepted this format 
```yaml
- example1: 
    bla bla bla
  example2: >
    bla bla bla
```
Above this one 
```yaml
[
    {
        "example1": "bla bla bla",
        "example2": "bla bla bla\n"
    }
]
```
as it gives line-length issue. So my vote to increase line length value.

File: rules/falco-deprecated_rules.yaml (hunk 2 above)
Comment (leogr) [reply rules/falco-deprecated_rules.yaml:150]: @incertum @h4l0gen 

I prefer _plain flow strings_ with no style modifiers  (i.e., ie no `>` or `|`) 

YAML block scalars delimited with style modifiers are intended to preserve (|) or fold (>) newlines. We don't need that. On the other hand, with plain strings, trailing `\n`s are automatically trimmed (that's the preferable, IMO). Do you agree?

You can find here some example: https://yaml-multiline.info/

So I definitively prefer:
```yaml
output:
```
(without `>`)

Comment (h4l0gen) [reply rules/falco-deprecated_rules.yaml:150]: Yes, I am removing `>` modifiers from all tags. I am going to push changes today with line-length <130, after that let see what other changes we feel.

**Comment by h4l0gen** [comment]
Hi @leogr, need your confirmation, which one you prefer 
```
- macro: inbound
  condition:
    (((evt.type in (accept,accept4,listen) and evt.dir=<) or
      (evt.type in (recvfrom,recvmsg) and evt.dir=< and
       fd.l4proto != tcp and fd.connected=false and fd.name_changed=true)) and
     (fd.typechar = 4 or fd.typechar = 6) and
     (fd.ip != "0.0.0.0" and fd.net != "127.0.0.0/8") and
     (evt.rawres >= 0 or evt.res = EINPROGRESS))
```
or 
```
- macro: inbound
  condition:
    (
     (
      (evt.type in (accept,accept4,listen) and evt.dir=<) or
      (evt.type in (recvfrom,recvmsg) and evt.dir=< and fd.l4proto != tcp and fd.connected=false and fd.name_changed=true)
     ) and
     (fd.typechar = 4 or fd.typechar = 6) and
     (fd.ip != "0.0.0.0" and fd.net != "127.0.0.0/8") and
     (evt.rawres >= 0 or evt.res = EINPROGRESS)
   )
```

**Comment by leogr** [comment]
> Hi @leogr, need your confirmation, which one you prefer
> 
> ```
> - macro: inbound
>   condition:
>     (((evt.type in (accept,accept4,listen) and evt.dir=<) or
>       (evt.type in (recvfrom,recvmsg) and evt.dir=< and
>        fd.l4proto != tcp and fd.connected=false and fd.name_changed=true)) and
>      (fd.typechar = 4 or fd.typechar = 6) and
>      (fd.ip != "0.0.0.0" and fd.net != "127.0.0.0/8") and
>      (evt.rawres >= 0 or evt.res = EINPROGRESS))
> ```
> 
> or
> 
> ```
> - macro: inbound
>   condition:
>     (
>      (
>       (evt.type in (accept,accept4,listen) and evt.dir=<) or
>       (evt.type in (recvfrom,recvmsg) and evt.dir=< and fd.l4proto != tcp and fd.connected=false and fd.name_changed=true)
>      ) and
>      (fd.typechar = 4 or fd.typechar = 6) and
>      (fd.ip != "0.0.0.0" and fd.net != "127.0.0.0/8") and
>      (evt.rawres >= 0 or evt.res = EINPROGRESS)
>    )
> ```

I prefer the first one.

**Pull request closed without merging**